core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -1421,15 +1422,15 @@ class LogManager(logDirs: Seq[File],
     for (dir <- liveLogDirs) {
       val cleanShutdownFileHandler = new CleanShutdownFileHandler(dir.getPath)
       val currentBrokerEpoch = cleanShutdownFileHandler.read
-      if (currentBrokerEpoch == -1L) {
+      if (!currentBrokerEpoch.isPresent) {
         info(s"Unable to read the broker epoch in ${dir.toString}.")
         return -1L
       }
-      if (brokerEpoch != -1 && currentBrokerEpoch != brokerEpoch) {
+      if (brokerEpoch != -1 && currentBrokerEpoch.getAsLong != brokerEpoch) {
         info(s"Found different broker epochs in ${dir.toString}. Other=$brokerEpoch vs current=$currentBrokerEpoch.")
         return -1L
       }
-      brokerEpoch = currentBrokerEpoch
+      brokerEpoch = currentBrokerEpoch.getAsLong
```

Comment:
Should we propagate Option[Long] all the way until we convert it to a request?

Reply:
Done.